### PR TITLE
Refactored sbt plugin

### PR DESCRIPTION
**Pull request by jroper**
* Moved everything to play.sbt package
* Introduced some deprecated vals and types in play package object
* Changed things like PlayRun and PlayCommands to be objects rather than traits
* Deleted unused code
* Removed IntelliJ support, IntelliJs built in sbt support is good enough, and people can manually use sbt-idea if they want

Of course, there's a lot more that needs to be refactored, but incremental improvements are best.

### CODE REVIEW

File: framework/src/sbt-plugin/src/main/scala/play/sbt/PlayCommands.scala [hunk 1]
```diff
@@ -67,19 +30,6 @@ trait PlayCommands extends PlayEclipse with PlayInternalKeys {
...
-  }
-
-  val shCommand = Command.args("sh", "<shell command>") { (state: State, args: Seq[String]) =>
-    if (args.isEmpty)
-      println("sh <command to run>")
-    else
-      fork(args)
-    state
-  }
-
```
Comment (jroper) [inline framework/src/sbt-plugin/src/main/scala/play/sbt/PlayCommands.scala]: I didn't even know this existed, I don't think it's at all useful, and it's the only thing that depended on `PlayJvm.scala`, so I deleted it.

File: framework/src/sbt-plugin/src/main/scala/play/sbt/PlayCommands.scala [hunk 2]
```diff
@@ -193,7 +143,6 @@ trait PlayCommands extends PlayEclipse with PlayInternalKeys {
     }.map(_.getCanonicalPath).distinct
   }
 
-  val computeDependencies = TaskKey[Seq[Map[Symbol, Any]]]("ivy-dependencies")
```
Comment (jroper) [inline framework/src/sbt-plugin/src/main/scala/play/sbt/PlayCommands.scala]: Moved to `PlayKeys`.

File: framework/src/sbt-plugin/src/main/scala/play/sbt/PlayExceptions.scala [hunk 3]
```diff
@@ -32,14 +32,4 @@ trait PlayExceptions {
...
   }
 
-  case class RoutesCompilationException(source: File, message: String, atLine: Option[Int], column: Option[Int]) extends PlayException.ExceptionSource(
-    "Compilation error", message) with FeedbackProvidedException {
-    def line = atLine.map(_.asInstanceOf[java.lang.Integer]).orNull
-    def position = column.map(_.asInstanceOf[java.lang.Integer]).orNull
-    def input = IO.read(source)
-    def sourceName = source.getAbsolutePath
-  }
-
```
Comment (jroper) [inline framework/src/sbt-plugin/src/main/scala/play/sbt/PlayExceptions.scala]: Moved to routes compiler.

File: framework/src/sbt-plugin/src/main/scala/play/sbtplugin/PlayPositionMapper.scala [hunk 4]
```diff
@@ -1,42 +0,0 @@
...
-
-import play.routes.compiler.RoutesCompiler
-import sbt._
-import xsbti.Position
-
-/**
- * Maps positions from compile errors on generated sources to positions in the original sources
- */
-trait PlayPositionMapper {
-  val routesPositionMapper: Position => Option[Position] = position => {
```
Comment (jroper) [inline framework/src/sbt-plugin/src/main/scala/play/sbtplugin/PlayPositionMapper.scala]: Moved to routes compiler.

**Comment by benmccann** [comment]
>[error]     x with query string
[error]      Got the exception com.google.inject.ProvisionException: Guice provision errors:
[error]      
[error]      1) Error in custom provider, com.typesafe.config.ConfigException$Missing: No configuration setting found for key 'stores'
[error]        while locating play.api.libs.ws.WSConfigParser
[error]        at play.api.libs.ws.ning.NingWSModule.bindings(NingWS.scala:520):
[error]      Binding(class play.api.libs.ws.WSClientConfig to ProviderConstructionTarget(class play.api.libs.ws.WSConfigParser) in interface javax.inject.Singleton)
[error]        while locating play.api.libs.ws.WSClientConfig
[error]          for parameter 0 at play.api.libs.ws.ning.NingWSClientConfigParser.<init>(NingConfig.scala:60)
[error]        at play.api.libs.ws.ning.NingWSClientConfigParser.class(NingConfig.scala:60)
[error]        while locating play.api.libs.ws.ning.NingWSClientConfigParser
[error]        at play.api.libs.ws.ning.NingWSModule.bindings(NingWS.scala:519):
[error]      Binding(class play.api.libs.ws.ning.NingWSClientConfig to ProviderConstructionTarget(class play.api.libs.ws.ning.NingWSClientConfigParser) in interface javax.inject.Singleton)
[error]        while locating play.api.libs.ws.ning.NingWSClientConfig
[error]          for parameter 1 at play.api.libs.ws.ning.NingWSAPI.<init>(NingWS.scala:531)
[error]        at play.api.libs.ws.ning.NingWSAPI.class(NingWS.scala:531)
[error]        while locating play.api.libs.ws.ning.NingWSAPI
[error]        while locating play.api.libs.ws.WSAPI
[error]      
[error]      1 error (WSSpec.scala:187)

**Comment by jroper** [comment]
That failure was caused by the issue that #4107 fixes.  @play-pull-request-validator

**Comment by jroper** [comment]
This is ready to merge.

File: framework/src/sbt-plugin/src/main/scala/PlayJvm.scala [hunk 5]
```diff
@@ -1,126 +0,0 @@
...
- */
-package play
-
-import sbt._
-import java.lang.{ ProcessBuilder => JProcessBuilder }
-import java.io.{ File => JFile, BufferedReader, InputStream, InputStreamReader, OutputStream }
-import scala.Console.{ GREEN, RESET }
-
-/**
- * based on https://github.com/typesafehub/sbt-multi-jvm
```
Comment (pvlugter) [inline framework/src/sbt-plugin/src/main/scala/PlayJvm.scala]: thought some of this looked familiar

File: framework/src/sbt-plugin/src/main/scala/play/sbt/PlayCommands.scala (hunk 1 above)
Comment (pvlugter) [inline framework/src/sbt-plugin/src/main/scala/play/sbt/PlayCommands.scala]: Makes sense.

File: framework/src/sbt-plugin/src/main/scala/play/sbtplugin/PlayPositionMapper.scala (hunk 4 above)
Comment (pvlugter) [inline framework/src/sbt-plugin/src/main/scala/play/sbtplugin/PlayPositionMapper.scala]: Much better.

**Comment by pvlugter** [comment]
Nice work. LGTM.

**Pull request merged**